docs/proposal/2025-02-24-rosa-networking.md

```diff
@@ -0,0 +1,128 @@
+
+All of the above can be currently provisioned and configured via AWS CLI, AWS Management Console or Terraform. Motivation for this work is to be able to provision and configure the network infrastructure for ROSA HCP using CAPI.
+
+### Goals
+
+1. Implement a namespaced new custom resource `RosaNetwork` representing the networking stack for ROSA HCP.
+2. It will be possible to reference the new `RosaNetwork` resource from ROSA control plane resource
+3. Implement creation and deletion for the new `RosaNetwork` resource.
+4. Support the same networking scenarios as [ROSA CLI](https://github.com/openshift/rosa) using the same embeded AWS CloudFormation template that ROSA CLI uses.
+
+### Non-Goals/Future Work
+
+- Modify current networking code in AWS / EKS clusters.
+- Support custom CloudFormation template.
+
+## Proposal
+
+The goal of this proposal is to be able to provision the networking infrastructure required for a ROSA HCP cluster.
+
+[ROSA CLI](https://github.com/openshift/rosa) supports creation of the networking infrastructure for ROSA HCP and uses [AWS CloudFormation](https://aws.amazon.com/cloudformation/) template under the hood. The [CloudFormation template used by ROSA CLI](https://github.com/openshift/rosa/blob/master/cmd/create/network/templates/rosa-quickstart-default-vpc/cloudformation.yaml) allows to specify four parameters: CloudFormation stack name, AZ count (between 1 and 3), region and CIDR block for the VPC. The created CloudFormation stack then contains a VPC, public and private subnets (each pair created in separate AZ), internet gateway attached to VPC, elastic IPs, NAT gateways, public and private routing tables and a security group. 
+
+Adopting the CloudFormation template used by rosa-cli would mean that CAPA and the `RosaNetwork` custom resource would be relying on a mechanism that is know to work well and any changes or fixes implemented in ROSA CLI would be picked up automatically in CAPA.
+
+In practical terms, implementation of the proposal would mean:
+1. A new namespaced custom resource definition `RosaNetwork` in CAPA with four attributes: name, AZ count, region and CIRD block for VPC. These four attributes will become the `spec` part of the new `RosaNetwork` type.
+
+1. A new reconciler for the new custom resource, implementing creation and deletion. The reconciler will be using an existing [CloudFormation template from ROSA CLI](https://github.com/openshift/rosa/blob/master/cmd/create/network/templates/rosa-quickstart-default-vpc/cloudformation.yaml) and will use AWS CloudFormation API to create and delete the AWS CloudFormation stack.
+
+   Outputs and resources created in the cloudformation stack will be tracked under `status` of the `RosaNetwork` type. In particular, the `status` will have the following fields:
+   * publicSubnets: array of public subnet ids
```

Comment:
make it one list of ;
- availabilityZone
  privateSubnet
  publicSubnet

Reply:
I don't understand the above. Do you want all the AZs, private and public subnets to be rolled into one array?

Reply:
Changed the `status` to be an array of the subnets grouped by AZs.